tests/unit/s2n_ktls_test.c

```diff
@@ -18,45 +18,90 @@
 #include "s2n_test.h"
 #include "testlib/s2n_testlib.h"
 
-S2N_RESULT s2n_ktls_retrieve_file_descriptor(struct s2n_connection *conn, s2n_ktls_mode ktls_mode, int *fd);
```

Comment:
I do like that you're no longer testing all your private methods instead of the "public" methods in your header. I think I prefer this version, but I can definitely see how that could be controversial :) Which version do you prefer?

Reply:
Forward declaring private methods is a pain (because of cascading api changes) and feels like a hack. Only testing the methods you expose in the API definitely feel cleaner.

The overhead of understanding that we override setsockopt and how we do it could be a ramp up for someone new but its more 'elegant' once you grasp it.